pkg/repository/manager/manager.go

```diff
@@ -32,42 +32,14 @@ import (
 
 	"github.com/vmware-tanzu/velero/internal/credentials"
 	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"github.com/vmware-tanzu/velero/pkg/repository"
 	"github.com/vmware-tanzu/velero/pkg/repository/provider"
 	"github.com/vmware-tanzu/velero/pkg/util/filesystem"
 	"github.com/vmware-tanzu/velero/pkg/util/kube"
 	"github.com/vmware-tanzu/velero/pkg/util/logging"
 	veleroutil "github.com/vmware-tanzu/velero/pkg/util/velero"
 )
 
-// SnapshotIdentifier uniquely identifies a snapshot
-// taken by Velero.
-type SnapshotIdentifier struct {
```

Comment:
If we only move `SnapshotIdentifier` to `types` folder, but keep `manager.go` as is in the `repository` folder, does that work?
The current changes look good to me, just curious why `manager.go` is moved as well.

Reply:
Per #8207

manager.go also import following which ultimately add cloud provider dependencies.

> due to manager.go import of
>
>pkg/repository/udmrepo/kopialib/ due to lib_repo.go && repo_init.go importing "github.com/kopia/kopia/*"
>pkg/repository/restic due to repository.go importing pkg/restic/common.go importing "github.com/vmware-tanzu/velero/pkg/repository/config" which have aws.go, azure.go etc...

Reply:
I'll double check again

Reply:
Moving only snapshotidentifier still result in dep in framework package.
pkg/plugin/framework/server.go depends on 
"github.com/vmware-tanzu/velero/pkg/cmd/server/config" where config.go depend on
"github.com/vmware-tanzu/velero/pkg/podvolume" which have files that import 
"github.com/vmware-tanzu/velero/pkg/repository" where manager.go imports
"github.com/vmware-tanzu/velero/pkg/repository/provider" which has imports to cloud deps via pkg/repository/provider/restic.go and pkg/repository/provider/unified_repo.go

So manager.go has to also move.. or refactor unified_repo.go & restic.go

Reply:
https://github.com/kaovilai/velero/tree/snapshotIdentifierOnly in case you wanna check. and run

```
❯ find ./pkg/plugin/framework/ -maxdepth 1 -type f | grep .go | xargs -I {} sh -c "echo && echo && echo {} && go list -f {{.Deps}} {} | sed 's/ /\n/g' | grep -E 'aws|azure|gcp|s3'"
```
while making changes. The only change there is only moving snapshotIdentifier. The deps issue isn't resolved.

Reply:
First of all, I agree on moving manager.go to a dedicate package.
Secondly, I will see it is not enough if the dependency chain is like this:
```
pkg/plugin/framework/server.go depends on
"github.com/vmware-tanzu/velero/pkg/cmd/server/config" where config.go depend on
"github.com/vmware-tanzu/velero/pkg/podvolume" which have files that import
```

`server/config` is a package to expose configs, but `podvolume` is a package to hold PVB/PVR code where uploader and repository packages must be imported there. So it is not rational to import such a heavy package just because of a config.
And as you can see why `podvolume` is required by `server/config`, it just because it requires definition of `podvolume.DefaultVolumesToFsBackup`.
Therefore, I think we need to remove the dependency of this:
```
"github.com/vmware-tanzu/velero/pkg/cmd/server/config" where config.go depend on
"github.com/vmware-tanzu/velero/pkg/podvolume" which have files that import
```

Reply:
ok sounds good.

Reply:
will see/try remove github.com/vmware-tanzu/velero/pkg/podvolume from `github.com/vmware-tanzu/velero/pkg/cmd/server/config`

Reply:
done. moved the default var to its own package `github.com/vmware-tanzu/velero/pkg/podvolume/defaults`
```
❯ find ./pkg/cmd/server/config -maxdepth 1 -type f | grep .go | xargs -I {} sh -c "echo && echo && echo {} && go list -f {{.Deps}} {} | sed 's/ /\n/g' | grep -E 'github.com/vmware-tanzu/velero/pkg/podvolume'" 


./pkg/cmd/server/config/config.go
github.com/vmware-tanzu/velero/pkg/podvolume/defaults
```